File: utils/csrc/udpsend.rb

require 'socket'
addr = ['<broadcast>', 6668]# broadcast address
#addr = ('255.255.255.255', 33333) # broadcast address explicitly [might not work ?]
#addr = ['127.0.0.255', 33333] # ??
UDPSock = UDPSocket.new
UDPSock.setsockopt(Socket::SOL_SOCKET, Socket::SO_BROADCAST, true)
data = 'foo';
UDPSock.send(data, 0, addr[0], addr[1])
UDPSock.close

File: utils/csrc/udpsend.pl

# retain for reference 

use warnings;
use strict;

use Socket qw(:all);
use POSIX ":sys_wait_h";

socket( SOCKET, PF_INET, SOCK_DGRAM, getprotobyname("udp") )
    or die "Error: can't create an udp socket: $!\n";

select( ( select(SOCKET), $|=1 )[0] ); # no from buffering

my $broadcastAddr = sockaddr_in(6668, INADDR_BROADCAST );
setsockopt( SOCKET, SOL_SOCKET, SO_BROADCAST, 1 );

send( SOCKET, "no_assigned", 0,  $broadcastAddr )
    or die "Error sending: $!\n";

close SOCKET;

